app/Application/Services/OpenWalletService.php

```diff
@@ -17,11 +17,14 @@ public function __construct(UserDataSource $userDataSource, WalletDataSource $wa
         $this->walletDataSource = $walletDataSource;
     }
 
+    /**
+     * @throws UserNotFoundException
+     */
     public function createWallet(string $userId): ?string
     {
         $user = $this->userDataSource->findById($userId);
         if (is_null($user)) {
-            return null;
+            throw new UserNotFoundException('User not found');
```

Comment:
What do you think to delete the message here and put it on the exception class??
```suggestion
            throw new UserNotFoundException();
```